src/foam/util/UIDGenerator.js

```diff
@@ -77,6 +96,31 @@ foam.CLASS({
         return permute(id.toString());
       `
     },
+    {
+      name: 'generate_',
+      args: [ 'StringBuilder id' ],
+      documentation: `
+      `,
+      javaCode: `
+        long curSec = 0;
+        int seqNo   = 0;
+
+        synchronized (this) {
+          curSec = (System.currentTimeMillis() - EPOCH) / 1000;
+          if ( curSec != getLastSecondCalled() ) {
+            setSeqNo(0);
+            setLastSecondCalled(curSec);
+          }
+          seqNo = getSeqNo();
+          setSeqNo(seqNo + 1);
+        }
+
+        // 2 bits timestamp
+        id.append(toHexString(curSec, getTimestampHexits()));
```

Comment:
the second arg of `toHexString(l, numberOfBytes)` is for adding leading zeros to the hex string, would not produce a smaller hex string of curSec.

Reply:
@lchanmann ok, so getNextLong handles overflow, so is there a cursec hexits length that works for both nuid and auid?